### PR TITLE
Fix i3bar leaks

### PR DESCRIPTION
**Pull request by orestisf1993**

### CODE REVIEW

File: i3bar/src/xcb.c [hunk 1]
```diff
@@ -1518,7 +1518,7 @@ void init_tray_colors(void) {
  */
 void clean_xcb(void) {
     i3_output *o_walk;
-    free_workspaces();
+    free_outputs();
     SLIST_FOREACH(o_walk, outputs, slist) {
```
Comment (Airblader) [inline i3bar/src/xcb.c]: Isn't this dead code now?

File: i3bar/src/ipc.c [hunk 2]
```diff
@@ -68,7 +68,6 @@ void got_output_reply(char *reply) {
     SLIST_FOREACH(o_walk, outputs, slist) {
         destroy_window(o_walk);
     }
-    FREE_SLIST(outputs, i3_output);
```
Comment (Airblader) [inline i3bar/src/ipc.c]: What happens to an output which has been removed in the reply? Won't it just stick around in `outputs` now?

Comment (orestisf1993) [reply i3bar/src/ipc.c]: `parse_outputs_json()` calls `free_outputs()`

File: i3bar/src/xcb.c (hunk 1 above)
Comment (orestisf1993) [reply i3bar/src/xcb.c]: yes

**Comment by orestisf1993** [comment]
up

**Comment by Airblader** [comment]
@orestisf1993 Can you take a look at the failing test?

**Comment by orestisf1993** [comment]
Should be ok now.

**Comment by Airblader** [comment]
Thanks!

**Pull request merged**